File: Android/demo/demo/src/main/java/com/didi/aoe/examples/demo/ShowcaseApp.java
package com.didi.aoe.examples.demo;

import android.app.Application;
import android.content.Context;

/**
 * @author noctis
 */
public class ShowcaseApp extends Application {
    private static Context mContext;

    public static Context getContext() {
        return mContext;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        mContext = getApplicationContext();

//        Aoe.init(getContext(), new AoeDataProvider() {
//            @Override
//            public long appId() {
//                return 164;
//            }
//
//            @Override
//            public double latitude() {
//                return 39.92;
//            }
//
//            @Override
//            public double longitude() {
//                return 116.46;
//            }
//        });
    }
}

File: Android/demo/demo/src/main/java/com/didi/aoe/examples/demo/ShowcaseApp.kt
package com.didi.aoe.examples.demo

import android.annotation.SuppressLint
import android.app.Application
import android.content.Context

/**
 * @author noctis
 */
class ShowcaseApp : Application() {

    override fun onCreate() {
        super.onCreate()
        context = applicationContext

        //        Aoe.init(getContext(), new AoeDataProvider() {
        //            @Override
        //            public long appId() {
        //                return 164;
        //            }
        //
        //            @Override
        //            public double latitude() {
        //                return 39.92;
        //            }
        //
        //            @Override
        //            public double longitude() {
        //                return 116.46;
        //            }
        //        });
    }

    companion object {
        @SuppressLint("StaticFieldLeak")
        var context: Context? = null
            private set
    }
}